src/sync.rs

```diff
@@ -679,6 +696,26 @@ impl<T: Copy> LockFreeFrozenVec<T> {
     }
 }
 
+impl<T: Copy + PartialEq> PartialEq for LockFreeFrozenVec<T> {
+    fn eq(&self, other: &Self) -> bool {
+        // first check the length
+        let self_len = self.len.load(Ordering::Acquire);
+        let other_len = other.len.load(Ordering::Acquire);
+        if self_len != other_len {
+            return false;
+        }
+
+        // since the lengths are the same, just check the elements in order
+        for index in 0..self_len {
```

Comment:
This is a TOCTOU violation

Reply:
ah, actually it is sound, but please add comments explaining why it is. "the indices are in bounds" is insufficient since the bounds may change, it's very important to note that the bounds can only increase.

Reply:
I can set the `locked` property for both if it helps.

Reply:
No, I think it's fine, please just write extensive safety comments

Reply:
Fixed in https://github.com/Manishearth/elsa/pull/62/commits/b9f7b83145640e4c8e01411ee781a582b836a01e